test_objectexplorer_collection_types gives lots of errors

<!--- **PLEASE READ:** When submitting here, please ensure you've completed the following checklist and checked the boxes to confirm. Issue reports without it may be closed. Thanks! --->

### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [x] Tried basic troubleshooting (if a bug/error)
    * [ ] Restarted Spyder
    * [ ] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [ ] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

Next on the list (after #17067) is `test_objectexplorer_collection_types` in `spyder/plugins/variableexplorer/widgets/objectexplorer/tests/test_objectexplorer.py`.  It seems like the current version of Python and maybe some packages leads to changes in the sizes of these objects.  Five of the parameter settings fail.

### What steps reproduce the problem?

```
(sid):~/debian/spyder-packages/spyder/build-area/spyder-5.2.0+dfsg1 $ CI=true xvfb-run -a -s "-screen 0 1024x768x24 +extension GLX" python3.9 -m pytest spyder/plugins/variableexplorer/widgets/objectexplorer/tests/test_objectexplorer.py
============================= test session starts ==============================
platform linux -- Python 3.9.9, pytest-6.2.5, py-1.10.0, pluggy-0.13.0
PyQt5 5.15.6 -- Qt runtime 5.15.2 -- Qt compiled 5.15.2
rootdir: /home/jdg/debian/spyder-packages/spyder/build-area/spyder-5.2.0+dfsg1, configfile: pytest.ini
plugins: typeguard-2.11.1, flaky-3.7.0, xvfb-2.0.0, mock-3.6.1, cov-3.0.0, order-1.0.0, qt-4.0.2, lazy-fixture-0.6.3
collected 9 items                                                              

spyder/plugins/variableexplorer/widgets/objectexplorer/tests/test_objectexplorer.py . [ 11%]
FFFF.F..                                                                 [100%]

=================================== FAILURES ===================================
________________ test_objectexplorer_collection_types[params0] _________________

objectexplorer = <function objectexplorer.<locals>.create_objectexplorer at 0x7f71880c3040>
params = ('kjkj kj k j j kj k jkj', [71, 78])

    @pytest.mark.parametrize('params', [
                # variable to show, rowCount for python 3 and 2
                ('kjkj kj k j j kj k jkj', [71, 78]),
                ([1, 3, 4, 'kjkj', None], [45, 46]),
                ({1, 2, 1, 3, None, 'A', 'B', 'C', True, False}, [54, 55]),
                (1.2233, [57, 58]),
                (np.random.rand(10, 10), [166, 162]),
                (datetime.date(1945, 5, 8), [43, 46])
            ])
    def test_objectexplorer_collection_types(objectexplorer, params):
        """Test to validate proper handling of collection data types."""
        test, row_count = params
        CONF.set('variable_explorer', 'show_special_attributes', True)
    
        # Editor was created
        editor = objectexplorer(test, name='variable')
        assert editor
    
        # Check number of rows and row content
        model = editor.obj_tree.model()
    
        # The row for the variable
        assert model.rowCount() == 1
    
        # Root row with children
        # Since rowCount for python 3 and 2 varies on differents systems,
        # we use a range of values
        expected_output_range = list(range(min(row_count), max(row_count) + 1))
>       assert model.rowCount(model.index(0, 0)) in expected_output_range
E       assert 80 in [71, 72, 73, 74, 75, 76, ...]
E        +  where 80 = <built-in method rowCount of TreeProxyModel object at 0x7f717f96a3a0>(<PyQt5.QtCore.QModelIndex object at 0x7f71858b1200>)
E        +    where <built-in method rowCount of TreeProxyModel object at 0x7f717f96a3a0> = <spyder.plugins.variableexplorer.widgets.objectexplorer.tree_model.TreeProxyModel object at 0x7f717f96a3a0>.rowCount
E        +    and   <PyQt5.QtCore.QModelIndex object at 0x7f71858b1200> = <built-in method index of TreeProxyModel object at 0x7f717f96a3a0>(0, 0)
E        +      where <built-in method index of TreeProxyModel object at 0x7f717f96a3a0> = <spyder.plugins.variableexplorer.widgets.objectexplorer.tree_model.TreeProxyModel object at 0x7f717f96a3a0>.index

spyder/plugins/variableexplorer/widgets/objectexplorer/tests/test_objectexplorer.py:136: AssertionError
________________ test_objectexplorer_collection_types[params1] _________________

objectexplorer = <function objectexplorer.<locals>.create_objectexplorer at 0x7f717f98a790>
params = ([1, 3, 4, 'kjkj', None], [45, 46])

    @pytest.mark.parametrize('params', [
                # variable to show, rowCount for python 3 and 2
                ('kjkj kj k j j kj k jkj', [71, 78]),
                ([1, 3, 4, 'kjkj', None], [45, 46]),
                ({1, 2, 1, 3, None, 'A', 'B', 'C', True, False}, [54, 55]),
                (1.2233, [57, 58]),
                (np.random.rand(10, 10), [166, 162]),
                (datetime.date(1945, 5, 8), [43, 46])
            ])
    def test_objectexplorer_collection_types(objectexplorer, params):
        """Test to validate proper handling of collection data types."""
        test, row_count = params
        CONF.set('variable_explorer', 'show_special_attributes', True)
    
        # Editor was created
        editor = objectexplorer(test, name='variable')
        assert editor
    
        # Check number of rows and row content
        model = editor.obj_tree.model()
    
        # The row for the variable
        assert model.rowCount() == 1
    
        # Root row with children
        # Since rowCount for python 3 and 2 varies on differents systems,
        # we use a range of values
        expected_output_range = list(range(min(row_count), max(row_count) + 1))
>       assert model.rowCount(model.index(0, 0)) in expected_output_range
E       assert 47 in [45, 46]
E        +  where 47 = <built-in method rowCount of TreeProxyModel object at 0x7f717430de50>(<PyQt5.QtCore.QModelIndex object at 0x7f71742cb510>)
E        +    where <built-in method rowCount of TreeProxyModel object at 0x7f717430de50> = <spyder.plugins.variableexplorer.widgets.objectexplorer.tree_model.TreeProxyModel object at 0x7f717430de50>.rowCount
E        +    and   <PyQt5.QtCore.QModelIndex object at 0x7f71742cb510> = <built-in method index of TreeProxyModel object at 0x7f717430de50>(0, 0)
E        +      where <built-in method index of TreeProxyModel object at 0x7f717430de50> = <spyder.plugins.variableexplorer.widgets.objectexplorer.tree_model.TreeProxyModel object at 0x7f717430de50>.index

spyder/plugins/variableexplorer/widgets/objectexplorer/tests/test_objectexplorer.py:136: AssertionError
________________ test_objectexplorer_collection_types[params2] _________________

objectexplorer = <function objectexplorer.<locals>.create_objectexplorer at 0x7f71742e23a0>
params = ({False, 1, 2, 3, 'B', 'A', ...}, [54, 55])

    @pytest.mark.parametrize('params', [
                # variable to show, rowCount for python 3 and 2
                ('kjkj kj k j j kj k jkj', [71, 78]),
                ([1, 3, 4, 'kjkj', None], [45, 46]),
                ({1, 2, 1, 3, None, 'A', 'B', 'C', True, False}, [54, 55]),
                (1.2233, [57, 58]),
                (np.random.rand(10, 10), [166, 162]),
                (datetime.date(1945, 5, 8), [43, 46])
            ])
    def test_objectexplorer_collection_types(objectexplorer, params):
        """Test to validate proper handling of collection data types."""
        test, row_count = params
        CONF.set('variable_explorer', 'show_special_attributes', True)
    
        # Editor was created
        editor = objectexplorer(test, name='variable')
        assert editor
    
        # Check number of rows and row content
        model = editor.obj_tree.model()
    
        # The row for the variable
        assert model.rowCount() == 1
    
        # Root row with children
        # Since rowCount for python 3 and 2 varies on differents systems,
        # we use a range of values
        expected_output_range = list(range(min(row_count), max(row_count) + 1))
>       assert model.rowCount(model.index(0, 0)) in expected_output_range
E       assert 56 in [54, 55]
E        +  where 56 = <built-in method rowCount of TreeProxyModel object at 0x7f71742e20d0>(<PyQt5.QtCore.QModelIndex object at 0x7f71742df510>)
E        +    where <built-in method rowCount of TreeProxyModel object at 0x7f71742e20d0> = <spyder.plugins.variableexplorer.widgets.objectexplorer.tree_model.TreeProxyModel object at 0x7f71742e20d0>.rowCount
E        +    and   <PyQt5.QtCore.QModelIndex object at 0x7f71742df510> = <built-in method index of TreeProxyModel object at 0x7f71742e20d0>(0, 0)
E        +      where <built-in method index of TreeProxyModel object at 0x7f71742e20d0> = <spyder.plugins.variableexplorer.widgets.objectexplorer.tree_model.TreeProxyModel object at 0x7f71742e20d0>.index

spyder/plugins/variableexplorer/widgets/objectexplorer/tests/test_objectexplorer.py:136: AssertionError
________________ test_objectexplorer_collection_types[params3] _________________

objectexplorer = <function objectexplorer.<locals>.create_objectexplorer at 0x7f71742d8790>
params = (1.2233, [57, 58])

    @pytest.mark.parametrize('params', [
                # variable to show, rowCount for python 3 and 2
                ('kjkj kj k j j kj k jkj', [71, 78]),
                ([1, 3, 4, 'kjkj', None], [45, 46]),
                ({1, 2, 1, 3, None, 'A', 'B', 'C', True, False}, [54, 55]),
                (1.2233, [57, 58]),
                (np.random.rand(10, 10), [166, 162]),
                (datetime.date(1945, 5, 8), [43, 46])
            ])
    def test_objectexplorer_collection_types(objectexplorer, params):
        """Test to validate proper handling of collection data types."""
        test, row_count = params
        CONF.set('variable_explorer', 'show_special_attributes', True)
    
        # Editor was created
        editor = objectexplorer(test, name='variable')
        assert editor
    
        # Check number of rows and row content
        model = editor.obj_tree.model()
    
        # The row for the variable
        assert model.rowCount() == 1
    
        # Root row with children
        # Since rowCount for python 3 and 2 varies on differents systems,
        # we use a range of values
        expected_output_range = list(range(min(row_count), max(row_count) + 1))
>       assert model.rowCount(model.index(0, 0)) in expected_output_range
E       assert 59 in [57, 58]
E        +  where 59 = <built-in method rowCount of TreeProxyModel object at 0x7f71742d8040>(<PyQt5.QtCore.QModelIndex object at 0x7f71742f52e0>)
E        +    where <built-in method rowCount of TreeProxyModel object at 0x7f71742d8040> = <spyder.plugins.variableexplorer.widgets.objectexplorer.tree_model.TreeProxyModel object at 0x7f71742d8040>.rowCount
E        +    and   <PyQt5.QtCore.QModelIndex object at 0x7f71742f52e0> = <built-in method index of TreeProxyModel object at 0x7f71742d8040>(0, 0)
E        +      where <built-in method index of TreeProxyModel object at 0x7f71742d8040> = <spyder.plugins.variableexplorer.widgets.objectexplorer.tree_model.TreeProxyModel object at 0x7f71742d8040>.index

spyder/plugins/variableexplorer/widgets/objectexplorer/tests/test_objectexplorer.py:136: AssertionError
________________ test_objectexplorer_collection_types[params5] _________________

objectexplorer = <function objectexplorer.<locals>.create_objectexplorer at 0x7f71742e5ca0>
params = (datetime.date(1945, 5, 8), [43, 46])

    @pytest.mark.parametrize('params', [
                # variable to show, rowCount for python 3 and 2
                ('kjkj kj k j j kj k jkj', [71, 78]),
                ([1, 3, 4, 'kjkj', None], [45, 46]),
                ({1, 2, 1, 3, None, 'A', 'B', 'C', True, False}, [54, 55]),
                (1.2233, [57, 58]),
                (np.random.rand(10, 10), [166, 162]),
                (datetime.date(1945, 5, 8), [43, 46])
            ])
    def test_objectexplorer_collection_types(objectexplorer, params):
        """Test to validate proper handling of collection data types."""
        test, row_count = params
        CONF.set('variable_explorer', 'show_special_attributes', True)
    
        # Editor was created
        editor = objectexplorer(test, name='variable')
        assert editor
    
        # Check number of rows and row content
        model = editor.obj_tree.model()
    
        # The row for the variable
        assert model.rowCount() == 1
    
        # Root row with children
        # Since rowCount for python 3 and 2 varies on differents systems,
        # we use a range of values
        expected_output_range = list(range(min(row_count), max(row_count) + 1))
>       assert model.rowCount(model.index(0, 0)) in expected_output_range
E       assert 47 in [43, 44, 45, 46]
E        +  where 47 = <built-in method rowCount of TreeProxyModel object at 0x7f71742af430>(<PyQt5.QtCore.QModelIndex object at 0x7f717428d580>)
E        +    where <built-in method rowCount of TreeProxyModel object at 0x7f71742af430> = <spyder.plugins.variableexplorer.widgets.objectexplorer.tree_model.TreeProxyModel object at 0x7f71742af430>.rowCount
E        +    and   <PyQt5.QtCore.QModelIndex object at 0x7f717428d580> = <built-in method index of TreeProxyModel object at 0x7f71742af430>(0, 0)
E        +      where <built-in method index of TreeProxyModel object at 0x7f71742af430> = <spyder.plugins.variableexplorer.widgets.objectexplorer.tree_model.TreeProxyModel object at 0x7f71742af430>.index

spyder/plugins/variableexplorer/widgets/objectexplorer/tests/test_objectexplorer.py:136: AssertionError
=========================== short test summary info ============================
FAILED spyder/plugins/variableexplorer/widgets/objectexplorer/tests/test_objectexplorer.py::test_objectexplorer_collection_types[params0]
FAILED spyder/plugins/variableexplorer/widgets/objectexplorer/tests/test_objectexplorer.py::test_objectexplorer_collection_types[params1]
FAILED spyder/plugins/variableexplorer/widgets/objectexplorer/tests/test_objectexplorer.py::test_objectexplorer_collection_types[params2]
FAILED spyder/plugins/variableexplorer/widgets/objectexplorer/tests/test_objectexplorer.py::test_objectexplorer_collection_types[params3]
FAILED spyder/plugins/variableexplorer/widgets/objectexplorer/tests/test_objectexplorer.py::test_objectexplorer_collection_types[params5]
========================= 5 failed, 4 passed in 1.77s ==========================
```
The same occurs outside of a chroot without using `xvfb`.

### What is the expected output? What do you see instead?

I wonder whether the limits specified in the `row_count` parameters need updating, or whether this is revealing a genuine bug? The actual outputs were each just one or two larger than expected, though `params4` appears to have been successful.

## Versions
<!--- You can get this information from Help > About Spyder...
or (if Spyder won't launch) the "conda list" command
from the Anaconda Prompt/Terminal/command line. --->

* Spyder version: 5.2.0
* Python version: 3.9 and 3.10
* Qt version: 5.15.2
* PyQt version: 5.15.6+dfsg-1+b1
* Operating System name/version: Debian testing and unstable

### Dependencies
<!--- Please go to the menu entry Help > Dependencies,
press the Copy to clipboard button and paste below --->

```
# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  2.0.0 (OK)
cookiecutter >=1.6.0          :  1.7.3 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.0.2 (OK)
IPython >=7.6.0               :  7.27.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.0 (OK)
jellyfish >=0.7               :  0.8.9 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  23.3.0 (OK)
nbconvert >=4.0               :  6.1.0 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
parso >=0.7.0;<0.9.0          :  0.8.1 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.8.0 (OK)
pygments >=2.0                :  2.7.1 (OK)
pylint >=2.5.0                :  2.12.2 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.3.1;<1.4.0          :  1.3.2 (OK)
pylsp_black >=1.0.0           :  None (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.2.1 (OK)
qtawesome >=1.0.2             :  1.1.1 (OK)
qtconsole >=5.2.1;<5.3.0      :  5.2.1 (OK)
qtpy >=1.5.0                  :  1.9.0 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  58.2.0 (OK)
sphinx >=0.6.6                :  4.3.1 (OK)
spyder_kernels >=2.2.0;<2.3.0 :  2.2.0 (OK)
textdistance >=4.2.0          :  4.2.2 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.6 (OK)
xdg >=0.26                    :  0.27 (OK)
zmq >=17                      :  22.3.0 (OK)

# Optional:
cython >=0.21                 :  0.29.24 (OK)
matplotlib >=2.0.0            :  3.3.4 (OK)
numpy >=1.7                   :  1.21.4 (OK)
pandas >=1.1.1                :  1.1.5 (OK)
scipy >=0.17.0                :  1.7.1 (OK)
sympy >=0.7.3                 :  1.7.1 (OK)
```

--- a/spyder/plugins/variableexplorer/widgets/objectexplorer/tests/test_objectexplorer.py
+++ b/spyder/plugins/variableexplorer/widgets/objectexplorer/tests/test_objectexplorer.py
@@ -106,14 +106,14 @@ def test_objectexplorer(objectexplorer):
 
 
 @pytest.mark.parametrize('params', [
-            # variable to show, rowCount for python 3 and 2
-            ('kjkj kj k j j kj k jkj', [71, 78]),
-            ([1, 3, 4, 'kjkj', None], [45, 46]),
-            ({1, 2, 1, 3, None, 'A', 'B', 'C', True, False}, [54, 55]),
-            (1.2233, [57, 58]),
-            (np.random.rand(10, 10), [166, 162]),
-            (datetime.date(1945, 5, 8), [43, 46])
-        ])
+    # variable to show, rowCount for different Python 3 versions
+    ('kjkj kj k j j kj k jkj', [71, 80]),
+    ([1, 3, 4, 'kjkj', None], [45, 47]),
+    ({1, 2, 1, 3, None, 'A', 'B', 'C', True, False}, [54, 56]),
+    (1.2233, [57, 59]),
+    (np.random.rand(10, 10), [166, 162]),
+    (datetime.date(1945, 5, 8), [43, 47])
+])
 def test_objectexplorer_collection_types(objectexplorer, params):
     """Test to validate proper handling of collection data types."""
     test, row_count = params

